### DISCUSSION TITLE
Master / Detail

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#610] [Q&A (Help)] [answered]
Using Backpack is there a way to create a crud where i've a main model and child one?
For example Orders and OrderRows:

Like BKP demo, i'd like to click on "+" and see all order rows :)  
<img width="741" alt="bkp-master-detail" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/4bc51670-a357-48b0-836b-7e696acc63fb">

Thanks for help

**Comment by karandatwani92** [comment]
@blondie63 
Yes, you can.
How? It depends on your need.

Examples:
Let's assume you want to **show** child attributes. You can do this using a field/column which supports relationships.
Let's assume you want to **create/update** parent and child from the same form.
Again you would be using a field/column that supports the relationship.
And for such fields which don't support relationships, you can use [$this->crud->getCurrentEntry();](https://backpackforlaravel.com/docs/6.x/crud-api#getters) and set the field value.
And to alter the saving process, you can override [store](https://backpackforlaravel.com/docs/6.x/crud-operation-create#override-the-store-method) & [update](https://backpackforlaravel.com/docs/6.x/crud-operation-update#override-the-update-method) method.

For the detail row, the doc link is here https://backpackforlaravel.com/docs/6.x/crud-api#details-row

**Reply by tabacitu** [reply]
@karandatwani92 let's stop recommending people override their `store()` and `update()` methods. Using [model events](https://backpackforlaravel.com/docs/6.x/crud-operation-create#use-events-in-your-setup-method) you can achieve 99% of the same things, without overriding operation logic. So it's most of the benefits with none of the downsides (people still get updates when we bugfix the operations).

Plus, creating events in your field definitions is so super-intuitive - https://backpackforlaravel.com/docs/6.x/crud-operation-create#use-events-in-your-field-definition

**Comment by tabacitu** [comment, marked as answer]
@blondie63 like Karan said, you can completely customize what's inside the detail row. See the docs here - https://backpackforlaravel.com/docs/6.x/crud-api#details-row

In addition to that, I'd suggest you take a look at https://backpackforlaravel.com/docs/6.x/crud-how-to#hasmany-1-n-relationship-with-subform-to-create-update-and-delet-1 because what you have there is a HasMany, so your Create and Update forms could look [like this](https://demo.backpackforlaravel.com/admin/pet-shop/invoice/create):
<img width="1512" alt="CleanShot 2023-08-07 at 14 54 24@2x" src="https://github.com/Laravel-Backpack/community-forum/assets/1032474/5664847a-dd2f-4ba0-8053-9993a9537bcb">

Hope it helps.

**Reply by blondie63** [reply]
Hi @tabacitu this view is good for an inline edit but really i'd like to have only a list page whit grid and nested grid, like this:
<img width="1070" alt="nested-grids" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/95af8800-42a8-4dcf-96f2-30e0853051e5">

Do you is it possible with BKP6 ?

**Reply by tabacitu** [reply]
I think you can go 80-90% of the way there with the details row, yes.

If that doesn't suit you, you can always create a custom operation that fits your needs perfectly - https://backpackforlaravel.com/docs/6.x/crud-operations#creating-a-custom-operation-1

It doesn't have to be general, since it's just for your project.
